tests/services/test_rdm_service.py

```diff
@@ -127,3 +128,42 @@ def test_embargo_lift_with_error(running_app, search_clear, minimal_record):
     # Record should not be lifted since it didn't expire (until 3220)
     with pytest.raises(EmbargoNotLiftedError):
         service.lift_embargo(_id=record["id"], identity=superuser_identity)
+
+
+def test_search_sort_verified_enabled(
+    running_app,
+    uploader,
+    minimal_record,
+    search_clear,
+    monkeypatch,
+    verified_user,
+):
+    """Tests sort by 'is_verified' field, when enabled.
+
+    The flag "RDM_SEARCH_SORT_BY_VERIFIED" is monkeypatched (only modified for this test).
+    """
+    service = current_rdm_records.records_service
+    monkeypatch.setitem(running_app.app.config, "RDM_SEARCH_SORT_BY_VERIFIED", True)
+
+    # NV : non-verified
+    nv_user = uploader
+    # V  : verified
+    v_user = verified_user
+
+    # Create two records for two distinct users (non-verified and verified)
+    nv_draft = service.create(nv_user.identity, minimal_record)
+    assert nv_draft
+    nv_record = service.publish(id_=nv_draft.id, identity=nv_user.identity)
+    assert nv_record
+
+    v_draft = service.create(v_user.identity, minimal_record)
+    assert v_draft
+    v_record = service.publish(id_=v_draft.id, identity=v_user.identity)
+    assert v_record
+
+    res = service.search(nv_user.identity)
+    assert res.total == 2
+    hits = res.to_dict()["hits"]["hits"]
+
+    expected_order = [v_record.id, nv_record.id]
+    assert expected_order == [h["id"] for h in hits]
```

Comment:
Nice test! However, it might be that the default sorting is latest first, and your test will pass anyway.
If you want to be 100% correct in your test, I would:
1. Make sure that the `unverified` record is the one published the last
2. Set `RDM_SEARCH_SORT_BY_VERIFIED = False`, the default
3. Search **sorting by latest** and assert that `unverified` is first
4. Set `RDM_SEARCH_SORT_BY_VERIFIED = True`
5. Search **sorting by latest** and assert that `verified`, even if older, is first

Reply:
Thanks for the suggestion!